rust/src/io/object_store.rs

```diff
@@ -75,9 +77,13 @@ async fn build_s3_object_store(uri: &str) -> Result<Arc<dyn OSObjectStore>> {
 }
 
 async fn build_gcs_object_store(uri: &str) -> Result<Arc<dyn OSObjectStore>> {
+    // GCS enables cache for public buckets, we disable to improve consistency
+    let mut headers = HeaderMap::new();
+    headers.insert(CACHE_CONTROL, "no-cache".parse().unwrap());
```

Comment:
Does this impact on read or write? Any implication that you can see we hardcode the cache behavior on user's behavior?

Reply:
Does it mark the file as "uncached" even users want to cache it (outside of the lance library).